### PR TITLE
Customer group commands for edition and deletion

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Customer group commands for edition and deletion
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI tests green, only new behat commands not integrated yet so no UI impacts
| UI Tests          | Not relevant
| Fixed issue or discussion?     | ~
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

File: src/Adapter/Customer/Group/Repository/GroupRepository.php [hunk 1]
```diff
@@ -82,6 +85,38 @@ public function assertGroupExists(GroupId $groupId): void
...
+        );
+
+        return new GroupId($groupId);
+    }
+
+    /**
+     * @param int $customerGroupId
+     *
+     * @return int[]
+     */
```
Comment (tleon) [inline src/Adapter/Customer/Group/Repository/GroupRepository.php:101]: unecessary php doc I guess

File: src/Adapter/Customer/Group/Repository/GroupRepository.php [hunk 2]
```diff
@@ -82,6 +85,38 @@ public function assertGroupExists(GroupId $groupId): void
...
+    public function getAssociatedShopIds(int $customerGroupId): array
+    {
+        return $this->getObjectModelAssociatedShopIds($customerGroupId, CustomerGroup::class);
+    }
+
+    /**
+     * @param CustomerGroup $customerGroup
+     */
```
Comment (tleon) [inline src/Adapter/Customer/Group/Repository/GroupRepository.php:109]: This one is unecessary too but you need to add a phpdoc for the array
```suggestion
    /**
     * @param string[] $propertiesToUpdate
     */
```

File: src/Core/Domain/Customer/Group/Command/EditCustomerGroupCommand.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    public function getShopIds(): ?array
+    {
+        return $this->shopIds;
+    }
+
+    /**
+     * @param int[] $shopIds
+     *
+     * @return $this
+     */
```
Comment (tleon) [inline src/Core/Domain/Customer/Group/Command/EditCustomerGroupCommand.php:122]:
```suggestion
```

File: src/Core/Repository/AbstractMultiShopObjectModelRepository.php [hunk 4]
```diff
@@ -244,4 +244,111 @@ protected function deleteObjectModelFromShops(ObjectModel $objectModel, array $s
             );
         }
     }
+
+    /**
+     * @param int $id
+     * @param string $objectModelClassName
+     *
```
Comment (tleon) [inline src/Core/Repository/AbstractMultiShopObjectModelRepository.php:251]:
```suggestion
```

**Review by tleon — APPROVED** [review]
Only some not blocking phpdoc things ;)

**Review by nicosomb — APPROVED** [review]

File: src/Adapter/Customer/Group/Repository/GroupRepository.php (hunk 1 above)
Comment (jolelievre) [reply src/Adapter/Customer/Group/Repository/GroupRepository.php:101]: Actually it's to indicate that the array is an array of int no ShopId

File: src/Adapter/Customer/Group/Repository/GroupRepository.php (hunk 2 above)
Comment (jolelievre) [reply src/Adapter/Customer/Group/Repository/GroupRepository.php:109]: You're right I'll update this in the next PR

File: src/Core/Domain/Customer/Group/Command/EditCustomerGroupCommand.php (hunk 3 above)
Comment (jolelievre) [reply src/Core/Domain/Customer/Group/Command/EditCustomerGroupCommand.php:122]: Same here it's to indicate the type in the array

File: src/Core/Repository/AbstractMultiShopObjectModelRepository.php (hunk 4 above)
Comment (jolelievre) [reply src/Core/Repository/AbstractMultiShopObjectModelRepository.php:251]: Yeah actually the PHPDoc is mainly for the return type to specify what's in the array

**Pull request merged**